.github/workflows/mega-linter-for-runner.yml

```
@@ -41,133 +123,228 @@ jobs:
     name: MegaLinter
     runs-on: ubuntu-latest
 
-    # Give the default GITHUB_TOKEN write permission to commit and push, comment
-    # issues & post new PR; remove the ones you do not need
+    ##
+    # This job's `GITHUB_TOKEN` or your added `PAT` must have the following
+    # permissions enabled to `write` for MegaLinter to function.
+    #
+    # Always aim to provide as few permissions as possible for personal access
+    # tokens. By default, all permissions are set to read only. Write
+    # permissions must be set for the following scopes in the described
+    # circumstances:
+    #
+    #   To have MegaLinter write and update a linting report comment on Pull
+    #   Requests, the `pull-requests` and `issues` permissions must be `write`.
+    #
+    #   To enable `APPLY_FIXES`, the `pull-requests` and `contents` permissions
+    #   must be `write`.
+    #
+    # More info at:
+    # https://docs.github.com/en/actions/security-guides/automatic-token-authentication
+    #
     permissions:
       contents: write
       issues: write
       pull-requests: write
-    environment: # ml workflow addition
-      name: dev # ml workflow addition
+
+    ##
+    # Sets the runner environment name to `dev`. Specific to this workflow.
+    #
+    environment:
+      name: dev
 
     steps:
 
-      # Git Checkout
+      ##
+      # Clone the Repository for linting.
+      #
+      # More info at:
+      # https://github.com/actions/checkout
+      #
       - name: Checkout Code
         uses: actions/checkout@v4
         with:
           token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
 
-          # If you use VALIDATE_ALL_CODEBASE = true, you can remove this line to
-          # improve performance
+          ##
+          # A fetch depth of `0` pulls the entire Repository's history,
+          # Branches, and Tags. Limiting this to `1` (or commenting out the
+          # line, as this is the default) increases efficiency, but is only
+          # applicable when `VALIDATE_ALL_CODE_BASE = true`.
+          #
           fetch-depth: 0
 
-      # MegaLinter
+      ##
+      # Run MegaLinter.
+      #
+      # More info at:
+      # https://github.com/oxsecurity/megalinter
+      #
       - name: MegaLinter
-
-        # You can override MegaLinter flavor used to have faster performances
-        # More info at https://megalinter.io/flavors/
-        uses: oxsecurity/megalinter/flavors/javascript@beta # ml workflow change
-
         id: ml
 
-        # All available variables are described in documentation
-        # https://megalinter.io/configuration/
+        ##
+        # The full MegaLinter project can be over 8GB in size! This is due to it including every linter available for
+        # the full MegaLinter project. It is highly recommended to select a more refined "flavor" that better-matches
+        # your project by changing the image used here.
```

```suggestion
        ##
        # The full MegaLinter project can be over 8GB in size! This is due to it
        # including every linter available for the full MegaLinter project. It
        # is highly recommended to select a more refined "flavor" that better
        # matches your project by changing the image used here.
```
To match mega-linter-runner/generators/mega-linter/templates/mega-linter.yml and remove the hyphen of the template